packages/evm/test/registry/guardians-registry-staking.js

```diff
@@ -702,6 +702,223 @@ contract('GuardiansRegistry', ([_, guardian, anotherGuardian]) => {
         itHandlesStakesProperlyForDifferentRecipients(data)
       })
     })
+
+    context('when the guardian requests to activate the tokens and lock activation', () => {
+      const data = LOCK_ACTIVATION_DATA
+
+      beforeEach('allow lock manager', async () => {
+        await registry.changeLockManager(guardian, true, { from: governor })
+      })
+
+      const itHandlesStakesWithLockActivationProperlyFor = (recipient, amount, data) => {
+        it('creates the lock', async () => {
+          await registry.stakeFor(recipient, amount, data, { from })
+
+          const { amount: lockedAmount, total } = await registry.getActivationLock(recipient, guardian)
+          assertBn(lockedAmount, amount, 'locked amount does not match')
+          assertBn(total, amount, 'total locked amount does not match')
+        })
+
+        it('adds the staked amount to the active balance of the recipient', async () => {
+          const { active: previousActiveBalance, available: previousAvailableBalance, locked: previousLockedBalance, pendingDeactivation: previousDeactivationBalance } = await registry.balanceOf(recipient)
+
+          await registry.stakeFor(recipient, amount, data, { from })
+
+          const { active: currentActiveBalance, available: currentAvailableBalance, locked: currentLockedBalance, pendingDeactivation: currentDeactivationBalance } = await registry.balanceOf(recipient)
+          assertBn(previousActiveBalance.add(amount), currentActiveBalance, 'recipient active balances do not match')
+
+          assertBn(previousLockedBalance, currentLockedBalance, 'recipient locked balances do not match')
+          assertBn(previousAvailableBalance, currentAvailableBalance, 'recipient available balances do not match')
+          assertBn(previousDeactivationBalance, currentDeactivationBalance, 'recipient deactivation balances do not match')
+        })
+
+        it('does not affect the active balance of the current term', async () => {
+          const termId = await controller.getLastEnsuredTermId()
+          const currentTermPreviousBalance = await registry.activeBalanceOfAt(recipient, termId)
+
+          await registry.stakeFor(recipient, amount, data, { from })
+
+          const currentTermCurrentBalance = await registry.activeBalanceOfAt(recipient, termId)
+          assertBn(currentTermPreviousBalance, currentTermCurrentBalance, 'current term active balances do not match')
+        })
+
+        if (recipient !== from) {
```

Comment:
Should we also add a test for the locked amounts? The recipient should be the one gaining a lock but not the sender.